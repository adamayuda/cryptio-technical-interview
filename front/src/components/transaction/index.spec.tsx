import React from "react";
import { Transaction } from ".";
import { render } from "@testing-library/react";

describe("src/components/transaction/index.tsx", () => {
  test("Render Transaction component", () => {
    const transaction = {
      hash: "hash",
      amount: 0,
      time: 1234,
    };

    const renderedTransaction = render(
      <Transaction transaction={transaction} />,
    );
    expect(renderedTransaction.container).toMatchSnapshot();
  });
  test("Render Transaction component", () => {
    const transaction = {
      hash: "hash",
      amount: -90,
      time: 1234,
    };

    const renderedTransaction = render(
      <Transaction transaction={transaction} />,
    );
    expect(renderedTransaction.container).toMatchSnapshot();
  });
});
