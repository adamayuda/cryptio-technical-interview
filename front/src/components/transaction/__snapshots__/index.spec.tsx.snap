// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/transaction/index.tsx Render Transaction component 1`] = `
<div>
  <div
    class="transaction"
  >
    <div
      class="header"
    >
      <h3>
        hash
      </h3>
      <div
        class="amount green"
      >
        0
      </div>
    </div>
    <div
      class="date"
    >
      <time
        datetime="1234000"
      >
        1970/01/01 04:20
      </time>
    </div>
  </div>
</div>
`;

exports[`src/components/transaction/index.tsx Render Transaction component 2`] = `
<div>
  <div
    class="transaction"
  >
    <div
      class="header"
    >
      <h3>
        hash
      </h3>
      <div
        class="amount red"
      >
        -9.000000000000001e-7
      </div>
    </div>
    <div
      class="date"
    >
      <time
        datetime="1234000"
      >
        1970/01/01 04:20
      </time>
    </div>
  </div>
</div>
`;
